src/mail/SendMailModel.js

```diff
@@ -784,9 +784,20 @@ export class SendMailModel {
 		blockingWaitHandler: (TranslationKey | lazy<string>, Promise<any>) => Promise<any> = (_, p) => p): Promise<void> {
 		const attachments = (saveAttachments) ? this._attachments : null
 		const {_draft} = this
-		const savePromise = Promise.resolve(_draft == null
+
+		// Create new drafts for drafts edited from trash or spam folder
+		const doCreateNewDraft = _draft
+			? this._mailModel.getMailboxFolders(_draft)
+			      .then(folders => folders.filter(f => f.folderType === MailFolderType.TRASH || f.folderType === MailFolderType.SPAM))
+			      .then(trashAndMailFolders =>
+				      trashAndMailFolders.find(folder =>
+					      isSameId(folder._id, neverNull(this._mailModel.getMailFolder(getListId(_draft)))._id)) != null)
```

Comment:
```js
 .then(trashAndMailFolders => trashAndMailFolders.find(folder => isSameId(folder.mails, getListId(_draft)) != null))
```

Reply:
Thanks! I thought that there must be an easier way to check the folder, but didn't see this one.

Reply:
The way you wrote it is more explicit so it's a plus but it results in a pretty long thing